### PR TITLE
Table Panel: use react as the default table panel

### PR DESCRIPTION
**Pull request by ryantxu**
We would like the react panel to be the default "table" panel in grafana.  It should replace the existing angular table options... however that will be very complicated given the possible complexity of the configurations.

This PR takes a middle ground.  It:
1. renames existing angular table to "table-old"
1. renames the dev react panel to "table"
1. Adds a migration handler to table... if it used to be angular, migrate automaticaly to "table-old"
1. "table-old" is now deprecated and can not be added to a dashboard.  (Will keep working if it already exists though)

This takes a perhaps crazy approach of throwing an error that the redux action checks -- any better ideas are totally welcome!

The PR is mostly working except:
* going to edit mode after migration fails?

After this PR we should look at:
* better migrations -- which parts of existing column regex system map to FieldConfig?

### CODE REVIEW

**Comment by ryantxu** [comment]
@hugohaggmark -- can you take a look?  is the error approach reasonable?  it seemed like the best way to get from table plugin all the way back to the actions

Any idea what is going on when trying to get into edit mode?  seems like a dispatch loop?

File: public/app/features/dashboard/state/actions.ts [hunk 1]
```diff
@@ -120,7 +120,16 @@ export function initDashboardPanel(panel: PanelModel): ThunkResult<void> {
...
-      panel.pluginLoaded(plugin);
+      try {
+        panel.pluginLoaded(plugin);
+      } catch (err) {
+        if (err.changePanelType) {
+          console.log('Plugin loading requires chaning panel type:', err.changePanelType);
+          await dispatch(changePanelPlugin(panel, err.changePanelType));
+          return; // dispatch will send the required events
+        }
+        console.log('Error loading panel', err, panel);
```
Comment (hugohaggmark) [inline public/app/features/dashboard/state/actions.ts:131]: Maybe we should dispatch an error action here and return?

**Comment by torkelo** [comment]
> Adds a migration handler to table... if it used to be angular, migrate automaticaly to "table-old" 

Cannot this be handled in a simpler way with a DashboardMigration before the dashboard loads?

File: public/app/features/dashboard/state/actions.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard/state/actions.ts:131]: I think we can handle this cleaner in a DashboardMigrartion, not sure there will be a general use case for this for other (external plugins), maybe we can find a new type of hook/info in plugin.json if that is the case to solve this more generally for external plugins (but do not think we have that yet).

Comment (ryantxu) [reply public/app/features/dashboard/state/actions.ts:131]: 👍 I have never seen the DashboardMigrator!

File: public/app/plugins/panel/table/README.md [hunk 2]
```diff
@@ -6,4 +6,4 @@ The table panel is very flexible, supporting both multiple modes for time series
 
 Check out the [Table Panel Showcase in the Grafana Playground](http://play.grafana.org/dashboard/db/table-panel-showcase) or read more about it here:
 
-[https://grafana.com/docs/grafana/latest/features/panels/table_panel/](https://grafana.com/docs/grafana/latest/features/panels/table_panel/)
+[http://docs.grafana.org/reference/table_panel/](http://docs.grafana.org/reference/table_panel/)
```
Comment (ryantxu) [inline public/app/plugins/panel/table/README.md]: This seems to happen automatically.... for this README I think the modification is wrong

**Review by torkelo — APPROVED** [review]
Looks good now

**Pull request merged**